src/synapse_mcp/entities/base.py

```diff
@@ -0,0 +1,85 @@
+import synapseclient
+from typing import Dict, List, Any, Optional, Union
+
+class BaseEntityOperations:
+    """Base class for entity operations."""
+    
+    def __init__(self, synapse_client: synapseclient.Synapse):
+        """Initialize with a Synapse client."""
+        self.synapse_client = synapse_client
+    
+    def get_entity_by_id(self, entity_id: str) -> Dict[str, Any]:
+        """Get an entity by ID.
+
+        Args:
+            entity_id: The Synapse ID of the entity
+
+        Returns:
+            The entity as a dictionary
+        """
+        # IMPORTANT: Always keep downloadFile=False to avoid downloading file content
+        # when fetching entity metadata. This prevents unnecessary downloads and
+        # keeps the operation lightweight for metadata-only operations.
+        entity = self.synapse_client.get(entity_id, downloadFile=False)
```

Comment:
There is an opportunity to swap over to https://python-docs.synapse.org/en/stable/reference/experimental/sync/factory_operations/#synapseclient.operations.get - It would have some downstream affects that also need to be addressed like swapping the `format_entity` over to use the new `snake_case` attributes. That has the consequence that the functions to list children of the project/folder also need to be swapped as well since they all eventually call to `format_entity`.